libs/utils/src/extract_cdf_ui_strings.ts

```diff
@@ -0,0 +1,138 @@
+// TODO(kofi): Remove once extractors are implemented.
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
+import _ from 'lodash';
+
+import {
+  BallotDefinition,
+  ElectionStringKey,
+  LanguageCode,
+  UiStringsPackage,
+} from '@votingworks/types';
+import { assertDefined } from '@votingworks/basics';
+
+const SUPPORTED_LANGUAGES = new Set<string>(Object.values(LanguageCode));
+
+/**
+ * Sets the appropriate language strings in supported languages for the given
+ * internationalized CDF ballot content text.
+ */
+function setInternationalizedUiStrings(params: {
+  uiStrings: UiStringsPackage;
+  stringKey: ElectionStringKey;
+  entityId?: string;
```

Comment:
For my understanding, could you explain when `entityId` will be specified vs. not?

Reply:
Yeah, glad you asked - wasn't sure of the naming here. It's used as a sub-key for strings with multiple "instances" like `contestTitle` or `candidateName`, where the entity ID would be contest ID or candidate ID, respectively (vs `electionTitle` for example, where there's only one instance).

LMK if that makes any lightbulbs go off for you WRT naming. Was waffling between `stringSubKey` and `entityId`, but neither's too obvious. Kind of leaning `stringSubKey` now though, since it's a bit more explicit WRT what it's used for in the translation mapping.

Reply:
K, I think https://github.com/votingworks/vxsuite/pull/4003/commits/ceaf8caf918eca9bcf937745ad25a01182116383 improves readability a bit - now defining `stringKey` as either a single `ElectionStringKey` or a pair of `[key: ElectionStringKey, subKey: string]`

Reply:
Ah perfect, this clarifies a lot! And I accordingly like the changes introduced in https://github.com/votingworks/vxsuite/commit/ceaf8caf918eca9bcf937745ad25a01182116383